### PR TITLE
add developer guide and cleanup docs

### PR DESCRIPTION
**Pull request by woodser**
Resolves https://github.com/haveno-dex/haveno/issues/118

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,17 +42,23 @@ Haveno is a community-driven project. For it to be succesful it's fundamental to
...
+See [docs/installing.md](docs/installing.md)
 
-## Bounties
+## Contributing to Haveno
 
-To incentivize development and reward contributors we adopt a simple bounty system. Contributors may be awarded bounties after completing a task (resolving an issue). Take a look at the issues eligible for a bounty on the [dedicated Kanban board](https://github.com/orgs/haveno-dex/projects/2) or look for [issues labelled '💰bounty'](https://github.com/haveno-dex/haveno/issues?q=is%3Aissue+is%3Aopen+label%3A%F0%9F%92%B0bounty) in the main `haveno` repository. [Details and conditions for receiving a bounty](docs/bounties.md).
+We are looking for help building Haveno. There are many [tasks](https://github.com/haveno-dex/haveno/issues) to complete, and many tasks have bounties.
 
-## Running a local Haveno test network
+See the [developer guide](./docs/developer-guide.md) to get started developing for Haveno. At this stage, we're mostly looking for help building the backend and APIs.
```
Comment (erciccione) [inline README.md:53]:
```suggestion
See the [developer guide](docs/developer-guide.md) to get started developing for Haveno. At this stage, we're mostly looking for help building the backend and APIs.
```

To make it consistent with the other links.

File: docs/developer-guide.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+5. [Download](https://search.maven.org/search?q=g:org.projectlombok%20AND%20a:lombok&core=gav) the latest version of the lombok jar.
+6. Run lombok jar, e.g.: `java -jar ~/Downloads/lombok-1.18.22.jar`.
+7. Follow prompts to install lombok to your Eclipse installation.
+8. Restart Eclipse.
+9. File > Import... > Existing Gradle Project.
+10. Select the location of the haveno project to import, e.g. ~/git/haveno.
+11. Advance to finish importing haveno project.
+12. File > Import... > Existing Projects into Workspace.
+13. Select location of haveno-ui-poc project to import, e.g. ~git/haveno-ui-poc.
+14. Advance to finish importing haveno-ui-poc project.
```
Comment (erciccione) [inline docs/developer-guide.md:31]: Better move this section to a dedicated document. We already have a dedicated intelliJ doc; adding a guide for another IDE inside another document would be inconsistent. Furthermore, This is not something all devs will need, but only the users of this specific IDE.

A good solution could be to have a dedicated file for instructions for IDEs and move the content of `idea-import.md` there.

File: docs/developer-guide.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+5. Build Haveno: `make`
+6. Follow [instructions](https://github.com/haveno-dex/haveno-ui-poc#how-to-update-the-protobuf-client) to update the protobuf client in haveno-ui-poc.
+7. Add the corresponding typescript method(s) to [HavenoDaemon.tsx](https://github.com/haveno-dex/haveno-ui-poc/blob/master/src/HavenoDaemon.tsx) with clear and concise documentation.
+8. Add clean and comprehensive tests to [HavenoDaemon.test.tsx](https://github.com/haveno-dex/haveno-ui-poc/blob/master/src/HavenoDaemon.test.tsx), following existing patterns.
+9. Verify the tests with `npm run test -- -t 'my test'` to run tests by name and `npm test` to run all tests together.
+10. Open a pull request for review.
+
+## Trade Protocol
+
+For documentation on the trade protocol, see [trade protocol](./trade_protocol/trade-protocol.pdf).
```
Comment (erciccione) [inline docs/developer-guide.md:55]: Note that github doesn't support showing PDFs, might make sense to link to the markdown file instead.

```suggestion
For documentation on the trade protocol, see [trade protocol](trade_protocol/trade-protocol.pdf).
```

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -1,17 +1,23 @@
...
+
 - Pull requests should contain as many details as possible about what you are going to change and why. Avoid "title only" PRs, unless they are self-explanatory.
 - Pull requests should contain one single commit, unless it makes sense to have more. Please become familiar with git if you are not.
-- Pull requests won't be merged before 24 hours has passed. This timeframe will be extended when Haveno will have more active developers.
\ No newline at end of file
+- Pull requests won't be merged before 24 hours has passed. This timeframe will be extended when Haveno will have more active developers.
+
+## Developer guide
+
+See the [developer guide](developer-guide.md) to get started developing for Haveno.
```
Comment (erciccione) [inline docs/CONTRIBUTING.md:23]:
```suggestion
See the [developer guide](developer-guide.md) to get started developing Haveno.
```

**Review by erciccione — CHANGES_REQUESTED** [review]

File: docs/import-haveno.md [hunk 5]
```diff
@@ -1,4 +1,36 @@
...
+This document describes how to import Haveno into an integrated development environment (IDE).
+
+## Importing Haveno into Eclipse IDE
+
+The following steps describe how to import Haveno into Eclipse IDE for development. You can also develop using [IntelliJ IDEA](#importing-haveno-into-intellij-idea) or VSCode if you prefer.
+
+> Note: Use default values unless specified otherwise.
+
+1. If you haven't already, clone Haveno projects:
+    1. `cd ~/git/`
```
Comment (erciccione) [inline docs/import-haveno.md:12]: I would remove this step entirely. We can assume the dev will have their own environment set up where they will clone the two repositories.

**Review by erciccione — COMMENTED** [review]
https://github.com/haveno-dex/haveno/blob/master/docs/README.md needs to be updated as well, but i would be more inclined for removing it. It's just another file to maintain.

File: docs/developer-guide.md (hunk 2 above)
Comment (woodser) [reply docs/developer-guide.md]: Done.

**Review by erciccione — APPROVED** [review]

**Pull request merged**